src/js/trees.js

```diff
@@ -0,0 +1,100 @@
+require.scopes.trees = (function() {
+
+/**
+ * Create a Dynamic tree node.
+ * takes a function that returns true/false.
+ * this becomes part of the array that builds trees
+ *
+ * returns function
+ */
+function dynamicNode(func) {
+  return function (node) {
+    if (typeof node.next != 'undefined') {
```

Comment:
Still figuring out what to do if the node is already set. I think the functions should chain, the first one returning true, returns its node.